File: Problems & Solutions/B/Build the smallest/10 December 2022/Java/Build the smallest.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.io.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int N = Integer.parseInt(read.readLine());
            String str = read.readLine();
            Solution ob = new Solution();

            System.out.println(ob.buildLowestNumber(str,N));
        }
    }
}
// } Driver Code Ends

class Solution {
    static String buildLowestNumber(String str, int N) {
        int n = str.length();
        Stack<Character> mystack=new Stack<>();
        for(int i=0;i<str.length();i++)
        {
            char c=str.charAt(i);
            while(!mystack.isEmpty() && N>0 && mystack.peek()>c)
            {   
                mystack.pop();
                N-=1;
            }
            
            if(!mystack.isEmpty() || c!='0')
                mystack.push(c);
        }
        while(!mystack.isEmpty() && N-->0)
            mystack.pop();
        if(mystack.isEmpty())
            return "0";
        
        StringBuilder sb=new StringBuilder();
        sb.append(str);
        while(!mystack.isEmpty())
        {
            sb.setCharAt(n-1,mystack.peek());
            mystack.pop();
            n-=1;
        }
        return sb.substring(n);
    }
}

File: Problems & Solutions/B/Build the smallest/10 December 2022/Python/Build the smallest.py
class Solution:

    def buildLowestNumber(self, num,k):
        stack = []
        for n in num:
            while(stack and int(stack[-1])>int(n) and k):
                k-=1
                stack.pop()
            stack.append(n)
        while(k):
            stack.pop()
            k-=1
        if len(stack)==0:
            return "0"
        s=''
        for i in range(len(stack)):
            s+=stack[i]
        return str(int(s))

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':

    t = int(input())

    for _ in range(t):
        N = int(input())
        S = input()

        solObj = Solution()

        print(solObj.buildLowestNumber(S,N))
# } Driver Code Ends